activerecord/lib/active_record/relation/batches.rb

```diff
@@ -341,7 +341,13 @@ def batch_on_loaded_relation(relation:, start:, finish:, order:, batch_limit:)
 
         if start || finish
           records = records.filter do |record|
-            (start.nil? || record.id >= start) && (finish.nil? || record.id <= finish)
+            id = record.id
+
+            if order == :asc
+              (start.nil? || id >= start) && (finish.nil? || id <= finish)
+            else
```

Comment:
is it possible for a nil `order` to make it's way here? if so then this is treated as the default - but maybe it's not possible. either way seems preferable to be explicit.